### ISSUE TITLE
Group Normalization

### ISSUE THREAD

**Issue by sshkhrnwbie** [mlpack/mlpack#1364]
Hi devs,

I would like to implement the Group Normalization layer which was proposed recently by He et al in [this paper](https://arxiv.org/pdf/1803.08494.pdf). I think tensorflow, pytorch and keras-contrib already have working implementations. Also Instance Normalization is compared with in that paper and I think mlpack lacks an implementation of it too.

Should I go ahead with it ? Or should we wait till it becomes more mainstream ?

**Comment by zoq** [comment]
Hello @sshkhrnwbie, the result are quite neat, so please feel free to open a PR 😃 .

**Comment by sshkhrnwbie** [comment]
I will work on it over the weekend and send one in a few days.

**Comment by zoq** [comment]
Sounds good, let me know if you need any help. I'll close this issue, we can discuss anything further in the PR.
